src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -122,75 +140,212 @@ public static function get_core_data() {
+	public static function get_theme_data() {
 		if ( null === self::$theme ) {
 			$theme_json_data = self::read_json_file( self::get_file_path_from_theme( 'theme.json' ) );
 			$theme_json_data = self::translate( $theme_json_data, wp_get_theme()->get( 'TextDomain' ) );
 			self::$theme     = new WP_Theme_JSON( $theme_json_data );
-		}
 
-		if ( empty( $theme_support_data ) ) {
-			return self::$theme;
+			if ( wp_get_theme()->parent() ) {
+				// Get parent theme.json.
+				$parent_theme_json_data = self::read_json_file( self::get_file_path_from_theme( 'theme.json', true ) );
+				$parent_theme_json_data = self::translate( $parent_theme_json_data, wp_get_theme()->parent()->get( 'TextDomain' ) );
+				$parent_theme           = new WP_Theme_JSON( $parent_theme_json_data );
+
+				// Merge the child theme.json into the parent theme.json.
+				// The child theme takes precedence over the parent.
+				$parent_theme->merge( self::$theme );
+				self::$theme = $parent_theme;
+			}
 		}
 
 		/*
-		 * We want the presets and settings declared in theme.json
-		 * to override the ones declared via add_theme_support.
-		 */
+		* We want the presets and settings declared in theme.json
+		* to override the ones declared via theme supports.
+		* So we take theme supports, transform it to theme.json shape
+		* and merge the self::$theme upon that.
+		*/
+		$theme_support_data  = WP_Theme_JSON::get_from_editor_settings( get_default_block_editor_settings() );
 		$with_theme_supports = new WP_Theme_JSON( $theme_support_data );
 		$with_theme_supports->merge( self::$theme );
 
 		return $with_theme_supports;
 	}
 
 	/**
-	 * There are different sources of data for a site: core and theme.
+	 * Returns the CPT that contains the user's origin config
+	 * for the current theme or a void array if none found.
+	 *
+	 * It can also create and return a new draft CPT.
 	 *
-	 * While the getters {@link get_core_data}, {@link get_theme_data} return the raw data
-	 * from the respective origins, this method merges them all together.
+	 * @since 5.9.0
 	 *
-	 * If the same piece of data is declared in different origins (core and theme),
-	 * the last origin overrides the previous. For example, if core disables custom colors
-	 * but a theme enables them, the theme config wins.
+	 * @param bool  $should_create_cpt  Optional. Whether a new CPT should be created if no one was found.
+	 *                                  False by default.
+	 * @param array $post_status_filter Filter Optional. CPT by post status.
+	 *                                   ['publish'] by default, so it only fetches published posts.
+	 *
+	 * @return array Custom Post Type for the user's origin config.
+	 */
+	private static function get_user_data_from_custom_post_type( $should_create_cpt = false, $post_status_filter = array( 'publish' ) ) {
+		$user_cpt         = array();
+		$post_type_filter = 'wp_global_styles';
+		$recent_posts     = wp_get_recent_posts(
```

Comment:
Please use `WP_Query` not `wp_get_recent_posts` here. We want to ensure that `'suppress_filters' => false,`